gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/mfa/MFAChallengeEndpoint.java

```diff
@@ -694,6 +694,7 @@ private void updateCredential(HttpServerRequest request, String credentialId, St
     }
 
     private void updateStrongAuthStatus(RoutingContext ctx) {
+        ctx.session().put(ConstantKeys.AUTH_COMPLETED, true);
```

Comment:
if it is Strongly authenticated it should be authenticated as well. The challenge tests are not passing without this because isValidSesion is always false.